src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2026,7 +2027,67 @@ protected override void HandleEvaluateRequestAsync(IRequestResponder<EvaluateArg
                 Type = variable.Type,
                 VariablesReference = variable.VariablesReference
             });
+        }
+
+
+        protected override void HandleReadMemoryRequestAsync(IRequestResponder<ReadMemoryArguments, ReadMemoryResponse> responder)
+        {
+            int hr;
+            ReadMemoryArguments rma = responder.Arguments;
+            ErrorBuilder eb = new ErrorBuilder(() => AD7Resources.Error_Scenario_ReadMemory);
+            try
+            {
+                if (string.IsNullOrEmpty(rma.MemoryReference))
+                {
+                    throw new ArgumentNullException("ReadMemoryArguments.MemoryReference is null or empty.");
+                }
 
+                ulong address;
+                if (rma.MemoryReference.ToLowerInvariant().StartsWith("0x"))
+                {
+                    address = Convert.ToUInt64(rma.MemoryReference.Substring(2), 16);
+                }
+                else
+                {
+                    address = Convert.ToUInt64(rma.MemoryReference, 10);
+                }
+
+                hr = ((IDebugMemoryBytesDAP)m_engine).CreateMemoryContext(address, out IDebugMemoryContext2 ppMemory);
+                eb.CheckHR(hr);
+
+                if (rma.Offset.HasValue && rma.Offset.Value != 0)
+                {
+                    ulong offset = Convert.ToUInt64(Math.Abs(rma.Offset.Value));
+                    if (rma.Offset > 0)
+                    {
+                        hr = ppMemory.Add(offset, out IDebugMemoryContext2 ppMemCxt);
+                    }
+                    else
+                    {
+                        hr = ppMemory.Subtract(offset, out IDebugMemoryContext2 ppMemCxt);
+                    }
+                    eb.CheckHR(hr);
+                }
+
+                byte[] data = new byte[rma.Count];
+                uint pdwUnreadable = 0;
+                hr = m_program.GetMemoryBytes(out IDebugMemoryBytes2 debugMemoryBytes);
+                eb.CheckHR(hr);
+
+                hr = debugMemoryBytes.ReadAt(ppMemory, (uint)rma.Count, data, out uint pdwRead, ref pdwUnreadable);
+                eb.CheckHR(hr);
```

Comment:
VS will sometimes request zero-byte reads just to convert an address+offset into an address. Might be worth an if here to skip this part of `rma.Count` is zero